protocol/communities/community.go

```diff
@@ -255,6 +256,158 @@ func (o *Community) MarshalPublicAPIJSON() ([]byte, error) {
 	return json.Marshal(communityItem)
 }
 
+func (o *Community) MarshalJSONWithMediaServer(mediaServer *server.MediaServer) ([]byte, error) {
```

Comment:
This is significant amount of duplication. Are we sure the desktop team will be able to transition to the new endpoint soon? Otherwise it's a large amount of duplication lying around.

A quick solution since both marshal functions are so similar is to create a common private function and parameterize what's different. But if the desktop team can adjust to the new endpoint soon(ish) than it's totally fine.

Here's a diff showing their similarity:

```diff
1c1
< func (o *Community) MarshalJSON() ([]byte, error) {
---
> func (o *Community) MarshalJSONWithMediaServer(mediaServer *server.MediaServer) ([]byte, error) {
21c21
< 		Images                      map[string]images.IdentityImage      `json:"images"`
---
> 		Images                      map[string]string                    `json:"images"`
118c118,127
< 		communityItem.CommunityTokensMetadata = o.config.CommunityDescription.CommunityTokensMetadata
---
> 
> 		// update token meta image to url rather than base64 image
> 		var tokenMetadata []*protobuf.CommunityTokenMetadata
> 		for _, m := range o.config.CommunityDescription.CommunityTokensMetadata {
> 			copyM := proto.Clone(m).(*protobuf.CommunityTokenMetadata)
> 			copyM.Image = mediaServer.MakeCommunityDescriptionTokenImageURL(o.IDString(), copyM.GetSymbol())
> 			tokenMetadata = append(tokenMetadata, copyM)
> 		}
> 		communityItem.CommunityTokensMetadata = tokenMetadata
> 
125c134
< 			for t, i := range o.config.CommunityDescription.Identity.Images {
---
> 			for t := range o.config.CommunityDescription.Identity.Images {
127c136
< 					communityItem.Images = make(map[string]images.IdentityImage)
---
> 					communityItem.Images = make(map[string]string)
129c138
< 				communityItem.Images[t] = images.IdentityImage{Name: t, Payload: i.Payload}
---
> 				communityItem.Images[t] = mediaServer.MakeCommunityImageURL(o.IDString(), t)
```

Reply:
> This is significant amount of duplication. Are we sure the desktop team will be able to transition to the new endpoint soon? Otherwise it's a large amount of duplication lying around.

We already had a task for 2.30 to addresses this https://github.com/status-im/status-desktop/issues/13061
It was a low priority task. However, since most of the hard work is done here, it should be a pretty quick win now to do it in Desktop.

Ping me when you have a mobile PR. We should be able to work on it soon-ish.

Reply:
Awesome, I'll schedule the implementation for next sprint

Reply:
Issue opened here https://github.com/status-im/status-desktop/issues/15340

Reply:
Added a PR for this:
https://github.com/status-im/status-go/pull/5459